server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -147,6 +154,10 @@ public void afterRefresh(boolean didRefresh) {
                                     final long lastRefreshedCheckpoint = ((InternalEngine) indexShard.getEngine())
                                         .lastRefreshedCheckpoint();
                                     indexShard.getEngine().translogManager().setMinSeqNoToKeep(lastRefreshedCheckpoint + 1);
+
+                                    if (!RemoteStoreSegmentUploadNotifier.NOOP.equals(uploadNotifier)) {
+                                        uploadNotifier.notifySegmentUpload(indexShard, checkpoint);
+                                    }
```

Comment:
can we inject the noop notifier and remove this condition. The noop notifier anyways would be harmless if it runs.